src/google/protobuf/compiler/objectivec/objectivec_helpers.cc

```diff
@@ -531,6 +539,11 @@ std::string FileClassPrefix(const FileDescriptor* file) {
     return file->options().objc_class_prefix();
   }
 
+  // If prefix is passed through objc_opt then accept that first.
+  if (!g_prefix_mode.objc_class_prefix().empty()) {
```

Comment:
So this test means there's no way to use the generation option set to the empty string, as it would appear to always be unset.  The on case that would come up is if someones build setup also ends up setting the support for use package prefix, so I'm not sure how critical it is, but figured I should mention in.

Maybe we should tweak where the option is collected and have that error if an empty string was passed?

Reply:
I agree. I can update to throw an error if the option is set with an empty string.